draft-ietf-core-sid.md

```diff
@@ -1308,6 +1314,12 @@ Assignment of SIDs to YANG items can be automated, the recommended process to as
 5. The "dependency-revision" should reflect the revision numbers of each
    YANG module that the YANG module imports at the moment of the generation.
 
+When updating a YANG module that is in active use, the existing SID assignments are maintained.
+(In contrast, when evolving an early draft that has not yet been adopted by a community of developers, SID assignments are often better done from scratch after a revision.)
+If the name of a schema node changes, but the data remain structurally and semantically similar to what was previously available under an old name, the SID that was used for the old name MAY continue to be used for the new name.
+If the meaning of an item changes, a new SID MAY be assigned to it; this is particular useful to allow the new SID to identify the new structure or semantics of the item.
```

Comment:
I would feel better if we could operationalize what "the encoding changes" actually means.
E.g., does going from uint64 to uint32 change the encoding?  (A big event in JSON-land.)
Converting a base type into a union is a non-event in XML, but a big change for YANG-CBOR because of YANG-XML's overlapping union idiosyncrasy.

Can we collect a few examples where YANG modules changed in time and decide whether the encoding changed?

Reply:
A data type change happens a lot for work-in-progress.
It is not supposed to happen after there is a published version.
The new SERVER work being adopted by NETMOD makes this more clear.

If the YANG data type changes in a new revision of a published module,
such that the CBOR encoding is changed, then a new SID SHOULD be assigned 


I agree it is difficult to make a hard rule.
There is precedent for treating published and unpublished modules differently.

Reply:
SEMVER not SERVER